### PR TITLE
SOC-2448 | fix: take into consideration removed categories

### PR DESCRIPTION
**Pull request by mleczey**
## Links

* https://wikia-inc.atlassian.net/browse/SOC-2448

## Description

As a user, I want to have my discussion category selections and sort order stored so they are set when I come back to that discussions site

## Reviewers

@bkoval

### CODE REVIEW

**Comment by bkoval** [comment]
How about we fix linter on this branch? :)

File: front/main/app/mixins/discussion-forum-actions-route.js [hunk 1]
```diff
@@ -9,9 +9,23 @@ export default Ember.Mixin.create(
...
+				params.catId = catId;
+				return params;
+			});
+		},
+
+		/**
+		 * @param {function} transform - function transforming query parameters, should return received query parameters
+		 */
+		updateStoredQueryParams(transform) {
+			const queryParams = localStorageConnector.getItem('discussionForumPreviousQueryParams');
```
Comment (bkoval) [inline front/main/app/mixins/discussion-forum-actions-route.js]: New line after const definitions :)

File: front/main/app/mixins/discussion-forum-actions-route.js [hunk 2]
```diff
@@ -9,9 +9,23 @@ export default Ember.Mixin.create(
...
+			if (queryParams) {
+				let params = JSON.parse(queryParams);
+				params = transform(params);
```
Comment (bkoval) [inline front/main/app/mixins/discussion-forum-actions-route.js]: what does this method do? :)

File: front/main/app/routes/discussion/forum.js [hunk 3]
```diff
@@ -72,11 +56,47 @@ export default DiscussionBaseRoute.extend(
...
+		 * @param {object} params
+		 */
+		transitionToPreviouslySelectedFilters(categories, params) {
+			if (localStorageConnector.getItem('discussionForumPreviousQueryParams')) {
+				this.validateAndUpdateStoredCategories(categories);
+
+				const previousParams =
+					JSON.parse(localStorageConnector.getItem('discussionForumPreviousQueryParams'));
+
+				if (params.catId.length === 0 && previousParams.catId.length > 0) {
```
Comment (bkoval) [inline front/main/app/routes/discussion/forum.js]: Should we check if catId exists first?

File: front/main/app/mixins/discussion-forum-actions-route.js (hunk 2 above)
Comment (mleczey) [inline front/main/app/mixins/discussion-forum-actions-route.js]: Method was described in jsdoc - parameter.

File: front/main/app/mixins/discussion-forum-actions-route.js (hunk 1 above)
Comment (mleczey) [inline front/main/app/mixins/discussion-forum-actions-route.js]: added

**Comment by bkoval** [comment]
LGTM

**Pull request merged**